### PR TITLE
Implement api,operation,rendering,draw:arguments:*

### PR DESCRIPTION
**Pull request by austinEng**
-----

- [x] New helpers, if any, are documented in `helper_index.md`.
- [x] Incomplete tests, if any, are marked with TODO or `.unimplemented()`.

**[Review requirement](https://github.com/gpuweb/cts/blob/main/docs/reviews.md) checklist:**

- [ ] WebGPU readability
- [ ] TypeScript readability

### CODE REVIEW

File: src/webgpu/api/operation/rendering/draw.spec.ts
Comment (Kangz) [inline src/webgpu/api/operation/rendering/draw.spec.ts:41]: nit: screen -> texture since we aren't using the swapchain.

Also maybe this test could be simpler if it were using points? WDYT?

File: src/webgpu/api/operation/rendering/draw.spec.ts
Comment (Kangz) [inline src/webgpu/api/operation/rendering/draw.spec.ts:66]: Maybe also test 1 instance? 0 instance means 0 triangles will be drawn. Ditto for count.

File: src/webgpu/api/operation/rendering/draw.spec.ts
Comment (Kangz) [inline src/webgpu/api/operation/rendering/draw.spec.ts:124]: It could be possible to use a larger render target to now have to mess with the best vertex in the shader.

File: src/webgpu/api/operation/rendering/draw.spec.ts
Comment (Kangz) [inline src/webgpu/api/operation/rendering/draw.spec.ts:140]: nit: `[[offset(0)]]` has been removed from WGSL

File: src/webgpu/api/operation/rendering/draw.spec.ts
Comment (Kangz) [inline src/webgpu/api/operation/rendering/draw.spec.ts:153]: nit: this could use the new render pipeline descriptor when webgpu/types is rolled to 0.0.44

**Review by Kangz — APPROVED** [review]
LGTM overall

File: src/webgpu/api/operation/rendering/draw.spec.ts
Comment (austinEng) [reply src/webgpu/api/operation/rendering/draw.spec.ts:41]: I think it's easier to test points in a test like this since we're translating from 0-1 or NDC coordinates to pixel offsets for the expectations. Triangles let us avoid getting the exact point and instead just check where the center of the triangle is.

File: src/webgpu/api/operation/rendering/draw.spec.ts
Comment (austinEng) [reply src/webgpu/api/operation/rendering/draw.spec.ts:66]: 0 is intentional. I think @kainino0x said we don't need to be testing 1 vs. non-zero here?

File: src/webgpu/api/operation/rendering/draw.spec.ts
Comment (austinEng) [reply src/webgpu/api/operation/rendering/draw.spec.ts:153]: This was before that landed, but I'll update this now.

**Comment by kainino0x** [comment]
Skipping reviewing this unless you think it needs a second reviewer

File: src/webgpu/api/operation/rendering/draw.spec.ts
Comment (austinEng) [reply src/webgpu/api/operation/rendering/draw.spec.ts]: Eh, added the cases for 1 triangle and 1 instance.

File: src/webgpu/api/operation/rendering/draw.spec.ts
Comment (austinEng) [reply src/webgpu/api/operation/rendering/draw.spec.ts]: Done.

File: src/webgpu/api/operation/rendering/draw.spec.ts
Comment (austinEng) [reply src/webgpu/api/operation/rendering/draw.spec.ts]: Done.

File: src/webgpu/api/operation/rendering/draw.spec.ts
Comment (austinEng) [reply src/webgpu/api/operation/rendering/draw.spec.ts:140]: Done.

File: src/webgpu/api/operation/rendering/draw.spec.ts
Comment (Kangz) [reply src/webgpu/api/operation/rendering/draw.spec.ts:41]: Ack

Comment (Kangz) [reply src/webgpu/api/operation/rendering/draw.spec.ts:41]: The nit: about screen->texture in the comment could still be addressed. LGTM otherwise.

**Review by Kangz — APPROVED** [review]

File: src/webgpu/api/operation/rendering/draw.spec.ts
Comment (austinEng) [inline src/webgpu/api/operation/rendering/draw.spec.ts:41]:
```suggestion
Horizontally across the texture are triangles with increasing "primitive id".
```

**Pull request merged**
